src/GovUk.Education.ExploreEducationStatistics.Data.Services/SubjectMetaService.cs

```diff
@@ -335,5 +333,114 @@ private static string OrderLocationAttributes(LocationAttributeNode node)
                 : await _releaseSubjectRepository.GetReleaseSubjectForLatestPublishedVersion(subjectId) ??
                   new Either<ActionResult, ReleaseSubject>(new NotFoundResult());
         }
+
+        private async Task<Either<ActionResult, Unit>> ValidateFiltersForSubject(
+            Guid subjectId,
+            List<FilterUpdateViewModel> requestFilters)
+        {
+            var filters = await _filterRepository.GetFiltersIncludingItems(subjectId);
+            return AssertCollectionsAreSameIgnoringOrder(filters,
+                requestFilters,
+                filter => filter.Id,
+                requestFilter => requestFilter.Id,
+                "Requested filters do not match subject filters").OnSuccess(_ =>
```

Comment:
Would you not want to add to the ValidationMessages Enum here to produce and error code rather than needing an plain English string here?  More machine readable for API clients to use?

Reply:
Oh cool - but I'm sorry, I thought that our ValidationResults could take arguments as well as the error code, whereas I see that I don't think we have that right now.  Happy if you wanted to back this out :grimacing: or equally happy for you to keep it as codes.

Reply:
I think it's cleaner with the codes rather than the strings so will keep it like it is for now. Being able to add arguments will be a good improvement for the future though!